diff --git a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dtsi
index e5c474e4d..695e23e3c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dtsi
@@ -48,6 +48,14 @@ &hdmi_receiver {
 	status = "okay";
 };
 
+&hdmi_receiver_sound {
+	status = "okay";
+};
+
+&i2s7_8ch {
+	status = "okay";
+};
+
 &pinctrl {
 	hdmirx {
 		hdmirx_hpd: hdmirx-5v-detection {
diff --git a/drivers/media/platform/synopsys/hdmirx/Kconfig b/drivers/media/platform/synopsys/hdmirx/Kconfig
index 4321f985f..9dd17f0ea 100644
--- a/drivers/media/platform/synopsys/hdmirx/Kconfig
+++ b/drivers/media/platform/synopsys/hdmirx/Kconfig
@@ -9,6 +9,7 @@ config VIDEO_SYNOPSYS_HDMIRX
 	select VIDEOBUF2_DMA_CONTIG
 	select CEC_CORE
 	select HDMI
+	select SND_SOC_HDMI_CODEC if SND_SOC
 	help
 	  Support for Synopsys HDMI HDMI RX Controller.
 	  This driver supports HDMI 2.0 version.
